scripts/manage_maintenance.py

```diff
@@ -0,0 +1,142 @@
+    add_parser.add_argument("--start", required=True,
+                            help="Event start time (ISO format)")
+    add_parser.add_argument("--duration", required=True,
+                            help="Event duration (e.g., '1h30m', '2d')")
+    add_parser.add_argument("--description", required=True,
+                            help="Event description")
+    add_parser.add_argument("--partial-responses", nargs="+",
+                            help="Catalog names for partial responses")
+    add_parser.add_argument("--degraded-performance", nargs="+",
+                            help="Catalog names for degraded performance")
+    add_parser.add_argument("--service-unavailable", nargs="+",
+                            help="Catalog names for service unavailability")
+    cancel_parser = subparsers.add_parser("cancel",
+                                          help="Cancel a pending event")
+    cancel_parser.add_argument("--index", type=int, required=True,
+                               help="Index of the event to cancel")
+    subparsers.add_parser("start", help="Activate a pending event")
+    subparsers.add_parser("end", help="Complete the active event")
+    adjust_parser = subparsers.add_parser("adjust",
+                                          help="Modify the active event")
+    adjust_parser.add_argument("--duration", required=True,
+                               help="New event duration (e.g., '1h30m', '2d')")
+
+    args = parser.parse_args(args)
+
+    service = MaintenanceService()
+
+    if args.command == "list":
+        events = service.get_schedule
+        if args.all:
+            events = events.to_json()
+        else:
+            active = events.active_event()
+            active = {} if active is None else {'active': active.to_json()}
+            pending = events.pending_events()
+            pending = {'pending': list(pe.to_json() for pe in pending)}
+            events = active | pending
+    elif args.command == "add":
+        duration = int(parse_duration(args.duration).total_seconds())
+        events = service.provision_event(planned_start=args.start,
+                                         planned_duration=duration,
+                                         description=args.description,
+                                         partial=args.partial_responses,
+                                         degraded=args.degraded_performance,
+                                         unavailable=args.service_unavailable)
+        events = service.add_event(events)
+    elif args.command == "cancel":
+        events = service.cancel_event(args.index)
+    elif args.command == "start":
+        events = service.start_event()
+    elif args.command == "end":
+        events = service.end_event()
+    elif args.command == "adjust":
+        events = service.adjust_event(parse_duration(args.duration))
+    else:
+        assert False, 'Invalid command'
```

Comment:
An `AssertionError` is not the proper exception to raise here. I believe this should be: 
```suggestion
        raise argparse.ArgumentError(subparsers, 'Invalid command')
```